scripts/dss_v2_apdapter.py

```diff
@@ -0,0 +1,489 @@
+        """
+        Validate and parse the given destination URL into bucket and path values.
+        """
+        split_url = parse.urlsplit(self.args.destination_url)
+        if not split_url.scheme == 'gs' or not split_url.netloc:
+            raise ValueError('Destination URL must be in gs://bucket/path format')
+        else:
+            path = split_url.path.strip('/')
+            return split_url.netloc, f'{path}/' if path else ''
+
+    def _get_bucket(self, bucket_name: str) -> Bucket:
+        """
+        Returns a reference to a GCS bucket.
+        """
+        log.info('Getting bucket: %s', bucket_name)
+        bucket = self.storage_client.get_bucket(bucket_name)
+        return bucket
+
+    def _run(self):
+        """
+        Request a list of all bundles in the DSS and process each bundle.
+        """
+        prefixes = self.get_prefix_list(self.args.bundle_uuid_prefix, self.args.bundle_uuid_start_prefix)
+
+        params = {
+            'replica': self.dss_src_replica,
+            'per_page': 500,  # limits how many bundles per request min=10 max=500
+        }
+        if prefixes:
+            params['prefix'] = prefixes.pop(0)
+        url_base = self.dss_endpoint + '/bundles/all'
+        url = url_base + '?' + parse.urlencode(params)
+
+        # Loop to request all pages of results for each prefix
+        while True:
+            log.info('Requesting list of bundles: %s', url)
+            response = requests.get(url)
+            response.raise_for_status()
+            response_json = response.json()
+            with ThreadPoolExecutor(max_workers=self.num_workers) as tpe:
+                future_to_bundle = {
+                    tpe.submit(self._process_bundle,
+                               bundle['uuid'],
+                               bundle['version']): f"{bundle['uuid']}.{bundle['version']}"
+                    for bundle in response_json['bundles']
+                }
+                for future in as_completed(future_to_bundle):
+                    bundle_fqid = future_to_bundle[future]
+                    try:
+                        future.result()
+                    except Exception as e:
+                        self.errors.append(f'Bundle {bundle_fqid} had Exception: {e}')
+                        f: Future
+                        for f in future_to_bundle:
+                            f.cancel()
+            if response_json.get('has_more'):
+                url = response_json.get('link')
+            elif prefixes:
+                params['prefix'] = prefixes.pop(0)
+                url = url_base + '?' + parse.urlencode(params)
+            else:
+                break
+
+    def _process_bundle(self, bundle_uuid: str, bundle_version: str):
+        """
+        Transfer all files from the given bundle to a GCS staging bucket.
+        """
+        if self.errors:
+            return
+        benchmarks = []
+        bundle_fqid = f'{bundle_uuid}_{bundle_version}'
```

Comment:
See, here you're inconsistent in the use of your separators. If you use BundleFQID, you will be getting a reasonably concise but consistent and self-explanatory string representation.